delta/plugins/storage/src/main/scala/ch/epfl/bluebrain/nexus/delta/plugins/storage/storages/Storages.scala

```diff
@@ -87,11 +87,11 @@ final class Storages private (
       id: IdSegment,
       projectRef: ProjectRef,
       source: Json
-  )(implicit caller: Caller): IO[StorageRejection, StorageResource] = {
+  )(implicit caller: Caller): IO[StorageResource] = {
     for {
-      pc            <- fetchContext.onCreate(projectRef)
-      iri           <- expandIri(id, pc)
-      storageFields <- sourceDecoder(projectRef, pc, iri, source)
+      pc            <- fetchContext.onCreate(projectRef).toCatsIO
+      iri           <- expandIri(id, pc).toCatsIO
```

Comment:
Could potentially reuse `FileId` in these methods - it encapsulates the Iri expansion and CE conversion. Might need a better name though 😅

Reply:
It's possible that we could have a broad trait (something suggesting a "full" id) and then specific classes like FileId extending it? I'm not sure whether to do that just for this though

Reply:
Yeah agreed - just working around `Resources` and a similar thing is done everywhere. I'll see what refactoring more generally would look like